src/anemoi/training/train/train.py

```diff
@@ -106,7 +105,7 @@ def initial_seed(self) -> int:
         (torch.rand(1), np_rng.random())
         LOGGER.debug(
             "Initial seed: Rank %d, initial seed %d, running with random seed: %d",
-            int(os.environ.get("SLURM_PROCID", "0")),
+            self.strategy.global_rank,
```

Comment:
is `self.strategy` guaranteed to be correctly initialized at this point?

Reply:
good point. I think it is since self.strategy is a `@cached_property` i.e. is initialized whenever first called

Reply:
🤔  yes, `Strategy.__init__` gets called when the property is first accessed, but I believe the internal state of the strategy (possibly including `self.strategy.global_rank`) isn't fully initialized until `strategy.setup()` gets called https://github.com/ecmwf/anemoi-training/blob/develop/src/anemoi/training/distributed/strategy.py#L44. And I reckon this doesn't happen until `trainer.fit(...)`?

https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/pytorch/trainer/trainer.py#L957

Reply:
you're right - it defaults to 0 before train() gets called. 

The idea was to get rid of all slurm dependencies but maybe in the case of this debug statement we could keep using `SLURM_PROCID`.

Reply:
I think the 0 default is acceptable in this case. The other ranks should log the correct value, after DDP init.